src/note.rs

```diff
@@ -169,14 +200,17 @@ impl Note {
     pub fn from_parts(
         recipient: Address,
         value: NoteValue,
+        asset: AssetBase,
         rho: Rho,
         rseed: RandomSeed,
     ) -> CtOption<Self> {
         let note = Note {
             recipient,
             value,
+            asset,
             rho,
             rseed,
+            rseed_split_note: CtOption::new(rseed, 0u8.into()),
```

Comment:
@str4d ([Link](https://github.com/zcash/orchard/pull/471#discussion_r2543527521)): ```suggestion
            rseed_split_note,
```